### PR TITLE
[AIRFLOW-6204] Add GCP system tests helper

### PR DESCRIPTION
**Pull request by turbaszek**
This PR adds GcpResourceHelper class that provides an cli for
creating GCP system tests. This feature allows a dveloper
to perform various actions of setting / tearing infrastructure.

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6204

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice. I love this refactor.

**Review by potiuk — COMMENTED** [review]
BTW. I think at some point in time those CLI commands can be added to the 'airflow' command (in development mode only). That would make all the system test commands available under one 'airflow systemtests' command for example.

**Comment by turbaszek** [comment]
@potiuk @mik-laj can you take a look before I refactor all system tests?

**Review by potiuk — APPROVED** [review]
❤️ Love it!

**Comment by turbaszek** [comment]
I decided to use argparse instead of click. Mainly because `click.command` decorator returns an object not the decorated function, thus decorated function can not be used in a pytes fixture.

**Comment by turbaszek** [comment]
I think I will split this PR into a few...

File: TESTING.rst
Comment (turbaszek) [inline TESTING.rst:127]: @mik-laj @potiuk @TobKed what do you think about this approach? Majority of tests will look like this. The problem appears when there's a lot of work to do in setup in tear down (that's why not touched cloud build or cloud sql). 

I can try to move "commons" to SystemTest class so everyone can reuse it (git, making files, etc). WDYT?

Comment (potiuk) [reply TESTING.rst:127]: I think it's really nice and clean. And It's ok to start with it and move more and more "commons" to system tests as we convert the system tests to follow the same pattern. We can gradually touch SystemTests in the same way. But I think we need a bit more explanation on how all this is going to be used:
1) how to use this in manual testing (what is the typical workflow of system tests executed by the developer - setup/running tests iteratively/teardown
2) how it looks like in automated system (basically just telling what are the steps executed one-by-one)

Ideally those should be based on examples :)

File: TESTING.rst
Comment (mik-laj) [inline TESTING.rst:89]:
```suggestion
Moreover, if a test requires time consuming setup or teardown it's recommend to add simple CLI. This helps a
```

File: tests/gcp/operators/test_cloud_memorystore_system.py
Comment (mik-laj) [inline tests/gcp/operators/test_cloud_memorystore_system.py:50]: I think we should wait with full migration on the pytest syntax until we abandon the development of the 1.10.x line.  Now there may be low awareness about how to write tests using pytest, so we cannot require knowledge of it. Such decision-making should be made at the level of the entire project and we should send links and materials that will allow for easy familiarization. pytest is not part of the standard library, so not everyone knows it.
https://lists.apache.org/thread.html/08b64d3b084c865399f98f6c6f56235ce5329e843d97938e1a8045a5%40%3Cdev.airflow.apache.org%3Ehttps://lists.apache.org/thread.html/08b64d3b084c865399f98f6c6f56235ce5329e843d97938e1a8045a5%40%3Cdev.airflow.apache.org%3E

Comment (potiuk) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: Why wait for abandoning development of 1.10.x? We already moved to pytest in 1.10 as well (I cherry-picked all the changes there and 1.10.7 already uses pytest). 

We also updated 1.10.7 documentation with information that we use pytest and that we should (for example) prefer standard python asserts which are overwritten by pytest. 

I don't think abandoning 1.10 line has anything to do with pytest. 

We already as community agreed to go pytest-first (there was vote about it).

Comment (potiuk) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: Here is the thread where community voted on pytest - with few "Huge +1"s, "Big +1"s , "Awesomes", "Fantastic", "Love pytest", "Can't waits" and not a single negative reaction... 

https://lists.apache.org/thread.html/484add15d555c717bab82553ba7e43a4e04f0d403ccf954406b382e7%40%3Cdev.airflow.apache.org%3E

Comment (mik-laj) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: I think this discussion should be on the mailing list.  In a moment I will create a new thread to discuss the topic of using new pytest features. Many people expressed their support for the question but did not comment on how to migrate the old convention to the new one. We don't have to use all the pytest features because the old code still works fine.

Comment (turbaszek) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: This change affects code that is rather not "common" and rather no one uses this. In my opinion fixtures are easier to read. I think that we already agreed that we do not touch old test but new ones can be pytest-like.

Comment (potiuk) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: Yep. I think that was the agreement with asserts (and it is even documented in https://github.com/apache/airflow/blob/master/TESTING.rst#writing-unit-tests). 

We have not discussed other features of pytest, but with universal acceptance of pytest I think it's just a matter of code review between committers if new tests are using some of pytest new features or not. And I think it would be pointless if after converting to pytest we should refrain from using some useful features. For example I wanted to (At the next opportunity) to start using parametrized from pytest- because it allows to re-run single tests from parametrized set of tests. It's super useful and we can have it in parallel to parameterized import. I wanted to try it out and if it works fine proposed to make it default.

I think this is a good idea when we start using new feature, agree (between reviewers) if it is better to use certain feature and before merging it just post a message to devlist that we are going to use it. I do not think we should have additional documentation/best practices etc. Pytes has already great documentation with many examples.

@mik-laj -> you did not explain why you thought we should wait for 1.10 abandoning? I think that was the proposal? What was the reasoning?

Comment (mik-laj) [reply tests/gcp/operators/test_cloud_memorystore_system.py:50]: Here is a thread on the mailing list https://lists.apache.org/thread.html/1e4df7d4b0cd9b2d2bb76a3336471aa85e852545dd41ada6d4e461b8%40%3Cdev.airflow.apache.org%3E

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Are we doing smth about it :)?

**Pull request closed without merging**
